server/src/main/kotlin/org/javacs/kt/externalsources/KlsURI.kt

```diff
@@ -54,19 +55,20 @@ data class KlsURI(val fileUri: URI, val query: Map<QueryParam, Any>) {
             ?.lastOrNull()
 
     val jarPath: Path
-        get() = Paths.get(fileUri.schemeSpecificPart.split("!")[0])
+        get() = Paths.get(parseURI(fileUri.schemeSpecificPart.split("!")[0]))
     val innerPath: String?
         get() = fileUri.schemeSpecificPart.split("!", limit = 2).get(1)
 
     val source: Boolean
-        get() = query[QueryParam.SOURCE] as? Boolean ?: false
+        get() = query[QueryParam.SOURCE].toString().toBoolean()
```

Comment:
I was thinking about this, perhaps we should use string values directly instead of (weakly typed) `Any` values and have consumers of the parameters deal with parsing, e.g.

```kotlin
get() = query[QueryParam.SOURCE]?.toBoolean() ?: false
```

in this case. That way we wouldn't need the `VALUE_PARSERS`, which are currently used inconsistently anyway.

Reply:
Agreed 👍 Updated the PR